src/main/java/Controller/CreateUser.java

```diff
@@ -0,0 +1,60 @@
+package Controller;
+
+import Helper.UniqueIdGenerator;
+import Entity.User;
+import DBAccesser.User.UserAccessor;
+import Helper.SuccessResponseGenerator;
+import Exceptions.UsernameAlreadyExistsException;
+import Exceptions.UserIdDoesNotExistException;
+import Exceptions.UsernameMissingFromRequestBodyException;
+
+import java.util.Map;
+import java.util.LinkedHashMap;
+import java.util.List;
+import javax.servlet.http.HttpServletRequest;  
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RequestBody;
+
+//this annotation tells that this class can contain methods which map to URL requests
+@RestController
+public final class CreateUser {
+
+    @Autowired 
+    private UserAccessor queryUser;
+
+    @Autowired
+    private UserAccessor insertUser;
+
+    @Autowired
+    private UniqueIdGenerator uniqueIdGenerator;
+    
+    @PostMapping("/users")
+    public Map<String, Object> createUser(@RequestBody Map<String, String> requestBody, HttpServletRequest request) {
+
+        String path = request.getRequestURI();
+
+        //check if request body is as required
+        if (!requestBody.containsKey("username")) {
+            throw new UsernameMissingFromRequestBodyException(path);
+        }
+
+        String username  = requestBody.get("username");
+
+        //check if username exists - return error if it does
+        if (queryUser.checkIfUsernameExists(username)) {
+            throw new UsernameAlreadyExistsException(path);
+        } 
+
+        User newUser = new User(username);
+        //generate unique userId
+        newUser.setUserId(uniqueIdGenerator.generateUniqueId("User", false, false));
```

Comment:
the later 2 arguments "false" and "false" are of no use, we can remove this from generateUniqueId as well. We can keep the Ids to be only positive, if the random number gives a negative number just apply modulus to it.

Reply:
Done